ddclient.in

```diff
@@ -5450,6 +5460,153 @@ sub nic_1984_update {
         }
     }
 }
+###################################################################### 
+## dinamico_cdmon_examples 
+###################################################################### 
+sub dinamico_cdmon_examples {
+return <<EoEXAMPLE;
+
+o 'cdmon'
+
+The 'cdmon' protocol is used by DNS services offered by cdmon.org.
+
+Configuration variables applicable to the 'cdmon' protocol are:
+  protocol=cdmon               ##
+  login=service-login          ## login name and password registered with the service
+  password=service-password    ##
+  fully.qualified.host         ## the host registered with the service.
+
+Example ${program}.conf file entries:
+  ## single host update
+  protocol=cdmon,                                               \\
+  login=my-my-cdmon.org-login,                                     \\
+  password=my-cdmon.org-password                                   \\
+  myhost.mydomain.cat
+
+EoEXAMPLE
+} 
+######################################################################
+## dinamico_cdmon_update
+##
+## adapted by David
+## https://ticket.cdmon.com/ca/support/solutions/articles/7000005922-api-d-actualitzaci%c3%b3-de-ip-del-dns-gratis-din%c3%a0mic?set_locale=1075
+## https://dinamico.cdmon.org/onlineService.php?enctype=MD5&n=usuario&p=1bc29b36f623ba82aaf6724fd3b16718&cip=x.x.x.x
+##
+######################################################################
+sub dinamico_cdmon_update {
+
+
+    debug("\ndinamico_cdmon_update -------------------");
+
+    ## update each configured host
+    foreach my $h (@_) {
+        my $ip = delete $config{$h}{'wantip'};
+        info("setting IP address to %s for %s", $ip, $h);
+        verbose("UPDATE:","updating %s", $h);
+
+        my $url;
+        $url   = "https://dinamico.cdmon.org/onlineService.php";
+        $url  .= "?enctype=MD5";
+        $url  .= "&n=$config{$h}{'login'}";
+        $url  .= "&p=$config{$h}{'password'}";
+        $url  .= "&cip=$ip" if $ip;
```

Comment:
```suggestion
        my $url = "https://dinamico.cdmon.org/onlineService.php?enctype=MD5&n=$config{$h}{'login'}&p=$config{$h}{'password'}";
        $url  .= "&cip=$ip" if $ip;
```

Can we use something other than MD5? MD5 is insecure for cryptography.

Reply:
It is the only thing that the provider accepts, the password must be in MD5